src/pywifes/wifes_wsol.py

```diff
@@ -1558,6 +1539,17 @@ def derive_wifes_optical_wave_solution(inimg,
   pmb = f[1].header.get('PMB')
   rh = f[1].header.get('RH')
   rma = f[1].header.get('ROTSKYPA') # Dumb name for rotator mechanical angle
+
+  # save some metadata to the return_dict. since this is always from the f[1] extension, a key without the slice `s` is sufficient
+  return_dict['grating'] = grating
+  return_dict['bin_x'] = bin_x
+  return_dict['bin_y'] = bin_y
+  return_dict['dateobs'] = dateobs
+  return_dict['tdk'] = tdk 
+  return_dict['pmb'] = pmb 
+  return_dict['rh'] = rh
+  return_dict['rma'] = rma

```

Comment:
Wound up with a few points in the one comment:

### 1:
Could declare & initialize the dictionary in one go here:

```
return_dict = {
  'grating': grating,
  etc.
}
```

### 2:
But if these values aren't changing, I'd say it's best to add everything at the end of the function instead.

In part, I think it's easiest to use the function if its return value is all defined in one place, so you can see all the headers.

### 3:
Might even be better to return a tuple instead of a dictionary:
```
return (grating, bin_x, bin_y, dateobs, etc.
```

If any fields aren't clear from the value, you could make them variables first:
```
slice_rk = new_r
return (..... slice_rk, slice_bla, etc.)
```

It'd save derive_wifes_optical_wave_solution from having to parse the key names, but it'd need to be careful with the positioning of its args:

```
(grating, bin_x, ..., slice_rk, slice_bla, ...) = run_slice(packaged_args)
```

Neither option is really ideal, but it's a private function, so it doesn't need to be perfectly structured. But given we don't currently have unit tests, I think a tuple is the most likely to (correctly) crash out if the private function was changed and the caller wasn't.